backstack/src/commonMain/kotlin/com/slack/circuit/backstack/Navigation.kt

```diff
@@ -18,16 +18,22 @@ package com.slack.circuit.backstack
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.Stable
 import androidx.compose.ui.Modifier
+import com.slack.circuit.runtime.screen.Screen
 import kotlinx.collections.immutable.ImmutableList
 
 /** Presentation logic for currently visible routes of a navigable UI. */
 @Stable
 public interface NavDecoration {
   @Composable
-  public fun <T> DecoratedContent(
+  public fun <T : NavArgument> DecoratedContent(
     args: ImmutableList<T>,
     backStackDepth: Int,
     modifier: Modifier,
     content: @Composable (T) -> Unit,
   )
 }
+
+/** Argument provided to [NavDecoration] that exposes the underlying [Screen]. */
+public interface NavArgument {
```

Comment:
what's the difference between this and accessing the screen via the `args` list (of records) that's passed in?

Reply:
Outside the specific implementation of `NavDecoration` we won't be able to access the specific screen as it'd end up being visible as `T`. Its needed for any animation override apis, not really for the current shared elements wok.